pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -740,3 +750,54 @@ func (ssr *Snapshotter) checkSnapstoreSecretUpdate() bool {
 	snapstoreHash[ssr.snapstoreConfig.Provider] = newSnapstoreSecretHash
 	return true
 }
+
+// IsFullSnapshotRequiredAtStartup checks whether to take a full snapshot or not during the startup of backup-restore.
+func (ssr *Snapshotter) IsFullSnapshotRequiredAtStartup(timeWindow float64) bool {
+	if ssr.PrevFullSnapshot == nil || ssr.PrevFullSnapshot.IsFinal || time.Since(ssr.PrevFullSnapshot.CreatedOn).Hours() > timeWindow {
+		return true
+	}
+
+	if !ssr.WasScheduledFullSnapshotMissed(timeWindow) {
+		return false
+	}
+	return ssr.IsNextFullSnapshotBeyondTimeWindow(timeWindow)
+}
+
+// WasScheduledFullSnapshotMissed determines whether the preceding full-snapshot was missed or not.
+func (ssr *Snapshotter) WasScheduledFullSnapshotMissed(timeWindow float64) bool {
+	now := time.Now()
+	nextSnapSchedule := ssr.schedule.Next(now)
+
+	ssr.logger.Infof(" previous schedule: %v", miscellaneous.GetPrevScheduledSnapTime(nextSnapSchedule, timeWindow))
+	if miscellaneous.GetPrevScheduledSnapTime(nextSnapSchedule, timeWindow) == ssr.PrevFullSnapshot.CreatedOn {
```

Comment:
Please call the function `miscellaneous.GetPrevScheduledSnapTime(nextSnapSchedule, timeWindow)` only once and assign to variable, rather than calling the same function with same input twice.

Reply:
yes, it was just for testing. I forgot to remove. thanks